### PR TITLE
Remove lifecycle node controllers

### PR DESCRIPTION
**Pull request by v-lopez**
Summary of changes:

- Replace `LifecyclePublisher`s with regular `Publisher`s, unfortunately now controllers could publish while deactivated.
- Refactor the calls to `get_lifecycle_node`

The `LifecyclePublisher` above made me notice that there are no LifecycleSubscribers or LifecycleServices. Is it because they have yet to be implemented, or because a `LifecycleNode` stops serving callbacks if it's not active?  I could not find mention to this.

### CODE REVIEW

File: joint_state_controller/src/joint_state_controller.cpp [hunk 1]
```diff
@@ -190,28 +188,24 @@ double get_value(
 controller_interface::return_type
 JointStateController::update()
 {
+  if (lifecycle_state_.id() != lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE) {
```
[the remark this answers is not in the document]
Comment (bmagyar) [reply joint_state_controller/src/joint_state_controller.cpp:191]: These are extra guarantees that were added, I don't think this was part of the lifecyclenode implementation in the first place.

**Review by bmagyar — APPROVED** [review]

File: joint_state_controller/src/joint_state_controller.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bmagyar) [reply joint_state_controller/src/joint_state_controller.cpp]: Moved to #127

File: joint_state_controller/src/joint_state_controller.cpp [hunk 2]
```diff
@@ -66,12 +67,15 @@ const
 rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
 JointStateController::on_configure(const rclcpp_lifecycle::State & /*previous_state*/)
 {
+  if (!node_.get()) {
```
[the remark this answers is not in the document]
Comment (bmagyar) [reply joint_state_controller/src/joint_state_controller.cpp]: Moved to #127

File: joint_state_controller/src/joint_state_controller.cpp [hunk 3]
```diff
@@ -23,6 +23,7 @@
 
 #include "hardware_interface/types/hardware_interface_return_values.hpp"
 #include "hardware_interface/types/hardware_interface_type_values.hpp"
+#include "lifecycle_msgs/msg/state.hpp"
```
Comment (destogl) [inline joint_state_controller/src/joint_state_controller.cpp:26]: Also remove this line please.

**Review by destogl — APPROVED** [review]
All good!

File: joint_state_controller/src/joint_state_controller.cpp (hunk 3 above)
Comment (v-lopez) [reply joint_state_controller/src/joint_state_controller.cpp]: Done

**Comment by bmagyar** [comment]
Ok I've touched things up a bit, added back the node nullptr check to facilitate one of the existing tests and modified another test to not try the ill-posed `update()` call. I think we should discuss what exactly the role of each state is, where we want to allow memory allocation and where is shouldn't be. When we mixed lifecycle nodes and the back-and-forth mechanism of "let me declare what I want then you assign" with controllers and CM made all of that a little fuzzy, the tests will probably need to be revised on a couple of places to reflect this properly. Not a blocker at the moment though.

**Pull request merged**